### ISSUE TITLE
Check for empty toc_buffer in initTOC

### ISSUE THREAD

**Issue by bikerspade** [superg/redumper#200]
If toc_buffer is empty, descriptors_count and descriptors may be invalid, leading to a segfault.

https://github.com/superg/redumper/blob/fc821074cfad5ece320c2e003d923469f8a06063/cd/toc.ixx#L819-L831

**Comment by superg** [comment]
When do you encounter such a situation?

TOC can't / shouldn't be empty, at least one track should be present. I can add a check to gracefully exit but it doesn't really make sense to "dump" the disc if there are no tracks.

**Comment by bikerspade** [comment]
In VGPC, a user experienced a segfault when running redumper for a burned CD in Linux:

Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000000004824e7 in gpsxre::TOC@cd.toc::initTOC(std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > const&) ()

**Comment by Deterous** [comment]
Also confirmed in Windows, so it is not Linux specific. It is a 716UF (USB model).

**Comment by superg** [comment]
Yeah I guess the disc track is not finalized and I don't really parse anything CD-R specific.
